velox/vector/FlatVector-inl.h

```diff
@@ -195,18 +222,32 @@ void FlatVector<T>::copyValuesAndNulls(
     }
     auto constant = source->asUnchecked<ConstantVector<T>>();
     T value = constant->valueAt(0);
-    rows.applyToSelected([&](int32_t row) { rawValues_[row] = value; });
+    if constexpr (std::is_same_v<T, bool>) {
+      auto range = rows.asRange();
+      auto* rawValues = reinterpret_cast<uint64_t*>(rawValues_);
+      if (value) {
+        bits::orBits(rawValues, range.bits(), range.begin(), range.end());
+      } else {
+        bits::andWithNegatedBits(
+            rawValues, range.bits(), range.begin(), range.end());
+      }
+    } else {
+      rows.applyToSelected([&](int32_t row) { rawValues_[row] = value; });
+    }
+
     rows.clearNulls(rawNulls);
   } else {
-    auto sourceVector = source->typeKind() != TypeKind::UNKNOWN
-        ? source->asUnchecked<SimpleVector<T>>()
-        : nullptr;
+    auto sourceVector = source->asUnchecked<SimpleVector<T>>();
     rows.applyToSelected([&](auto row) {
       auto sourceRow = toSourceRow ? toSourceRow[row] : row;
       if (!source->isNullAt(sourceRow)) {
```

Comment:
can be done as a follow up: was wondering if it makes sense to use a decoded vector here instead as `source->isNullAt` and `source->valueAt` would end up as virtual calls in the hot loop